pootle/apps/pootle_language/models.py

```diff
@@ -169,18 +169,17 @@ def get_cachekey(self):
     # # # /TreeItem
 
     def get_stats_for_user(self, user):
-        self.set_children(self.get_children_for_user(user))
-
+        self.set_children(
+            self.get_children_for_user(
+                user, select_related=["directory"]))
         return self.get_stats()
 
-    def get_children_for_user(self, user):
-        translation_projects = self.translationproject_set \
-                                   .for_user(user) \
-                                   .order_by('project__fullname')
-        user_tps = filter(lambda x: x.is_accessible_by(user),
-                          translation_projects)
-
-        return user_tps
+    def get_children_for_user(self, user, select_related=None):
+        tps = self.translationproject_set.for_user(
+            user, select_related=select_related)
+        return filter(
+            lambda x: x.is_accessible_by(user),
```

Comment:
`x.project.code` is retrieved in `x.is_accessible_by(user)` method. select_related(`project__code`) should be better.